addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/TestCrossSiteScriptV2.java

```diff
@@ -622,70 +679,78 @@ public void scan(HttpMessage msg, String param, String value) {
-                                                Alert.CONFIDENCE_MEDIUM,
-                                                null,
-                                                param,
-                                                ctx.getTarget(),
-                                                "",
-                                                ctx.getTarget(),
-                                                contexts2.get(0).getMsg());
-                                    } else {
-                                        HttpMessage ctx2Message = contexts2.get(0).getMsg();
-                                        if (StringUtils.containsIgnoreCase(
-                                                ctx.getMsg()
-                                                        .getResponseHeader()
-                                                        .getHeader(HttpHeader.CONTENT_TYPE),
-                                                "json")) {
-                                            bingo(
-                                                    Alert.RISK_LOW,
-                                                    Alert.CONFIDENCE_LOW,
-                                                    Constant.messages.getString(
-                                                            MESSAGE_PREFIX + "json.name"),
-                                                    Constant.messages.getString(
-                                                            MESSAGE_PREFIX + "json.desc"),
-                                                    ctx2Message
-                                                            .getRequestHeader()
-                                                            .getURI()
-                                                            .toString(),
-                                                    param,
-                                                    GENERIC_SCRIPT_ALERT,
-                                                    Constant.messages.getString(
-                                                            MESSAGE_PREFIX + "otherinfo.nothtml"),
-                                                    getSolution(),
-                                                    "",
-                                                    ctx2Message);
-                                        } else {
+                        for (String scriptAlert : GENERIC_SCRIPT_ALERT_LIST) {
+                            if (context.getMsg()
+                                    .getResponseBody()
+                                    .toString()
+                                    .contains(context.getTarget())) {
+                                List<HtmlContext> contexts2 =
+                                        performAttack(
+                                                msg, param, scriptAlert, null, 0, false, true);
+                                if (contexts2 == null) {
+                                    return;
```

Comment:
if breaked here the we might iterate the list of other contexts which is not needed as Stop event is called.